src/Microsoft.Framework.Runtime/PatternsCollectionHelper.cs

```diff
@@ -53,7 +58,12 @@ private static IEnumerable<string> CreateCollection(string projectDirectory, par
             {
                 if (Path.IsPathRooted(pattern))
                 {
-                    throw new InvalidOperationException(string.Format("Patten {0} is a rooted path, which is not supported.", pattern));
+                    throw new InvalidOperationException($"Pattern {pattern} is a rooted path, which is not supported.");
+                }
+
+                if (literalPath && pattern.Contains('*'))
+                {
+                    throw new InvalidOperationException($"Pattern {pattern} is not expected to contain wildcard characters.");
```

Comment:
It's a private static method and you have access to the value in all the immediate callers. It seems like a small amount of effort for a much clearer error. We can drop the second part (showing the wildcard property name) since that will require a more dramatic amount of refactoring.

Regardless of how well we illustrate where the error is, we still need to have a clear description of the problem.